clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -241,13 +242,30 @@ public void onSuccess(ClientResponse resp) {
 
                                 for (Map.Entry<TopicPartition, FetchResponse.PartitionData<Records>> entry : response.responseData().entrySet()) {
                                     TopicPartition partition = entry.getKey();
-                                    long fetchOffset = data.sessionPartitions().get(partition).fetchOffset;
-                                    FetchResponse.PartitionData<Records> fetchData = entry.getValue();
-
-                                    log.debug("Fetch {} at offset {} for partition {} returned fetch data {}",
-                                            isolationLevel, fetchOffset, partition, fetchData);
-                                    completedFetches.add(new CompletedFetch(partition, fetchOffset, fetchData, metricAggregator,
-                                            resp.requestHeader().apiVersion()));
+                                    FetchRequest.PartitionData requestData = data.sessionPartitions().get(partition);
+                                    if (requestData == null) {
+                                        String message;
+                                        if (data.metadata().isFull()) {
+                                            message = MessageFormatter.arrayFormat(
+                                                    "Response for missing full request partition: partition={}; metadata={}",
+                                                    new Object[]{partition, data.metadata()}).getMessage();
```

Comment:
Any reason why we didn't use String.format here?

Reply:
No technical reason. I see that `String.format` is used in a few places already. Let me know if you would like me to change this.

Reply:
@ijuma, Actually, I now remember why I did it. `MessageFormatter` supports Java arrays while `String.format` does not.

Reply:
Why would you want arrays instead of varargs though?

Reply:
```
String.format("array=%s", new Object[]{1,2,3})
array=[I@76f78e3a

MessageFormatter.arrayFormat("array={}", new Object[]{new Object[]{1,2,3}}).getMessage()
array=[1,2,3]
```
Right?